kinder/README.md

```diff
@@ -0,0 +1,257 @@
+| kubeadm-join    | Executes the kubeadm-join workflow both on secondary control plane nodes and on worker nodes. Available options are:<br /> `--use-phases` triggers execution of the init workflow by invoking single phases.<br />`--automatic-copy-certs` instruct kubeadm to use the automatic copy cert feature.<br /> `--only-node` to execute this action only on a specific node. |
+| kubeadm-upgrade |Executes the kubeadm upgrade workflow and upgrading K8s. Available options are:<br /> `--upgrade-version` for defining the target K8s version.<br />`--only-node` to execute this action only on a specific node.                                             |
+| Kubeadm-reset   | Executes the kubeadm-reset workflow on all the nodes. Available options are:<br />  `--only-node` to execute this action only on a specific node. |
+| cluster-info    | Returns a summary of cluster info including<br />- List of nodes<br />- list of pods<br />- list of images used by pods<br />- list of etcd members |
+| smoke-test      | Implements a non-exhaustive set of tests that aim at ensuring that the most important functions of a Kubernetes cluster work |
+
+### kinder exec
+
+`kinder exec` provide a topology aware wrapper on docker `docker exec` .
+
+```bash
+# check the kubeadm version on all the nodes
+kinder exec @all -- kubeadm version
+
+# run kubeadm join on all the worker nodes
+kinder exec @w* -- kubeadm join 172.17.0.2:6443 --token abcdef.0123456789abcdef ...
+
+# run kubectl command inside the bootstrap control-plane node
+kinder exec @cp1 -- kubectl --kubeconfig=/etc/kubernetes/admin.conf cluster-info
+```
+
+Following node selectors are available
+
+| selector | return the following nodes                                   |
+| -------- | ------------------------------------------------------------ |
+| @all     | all the Kubernetes nodes in the cluster.<br />(control-plane and worker nodes are included, load balancer and etcd not) |
+| @cp*     | all the control-plane nodes                                  |
+| @cp1     | the bootstrap-control plane node                             |
+| @cpN     | the secondary master nodes                                   |
+| @w*      | all the worker nodes                                         |
+| @lb      | the external load balancer                                   |
+| @etcd    | the external etcd                                            |
+
+As alternative to node selector, the node name (the container name without the cluster name prefix) can be used to target actions to a specific node.
+
+```bash
+# run kubeadm join on the first worker node node only
+kinder exec worker1 -- kubeadm join 172.17.0.2:6443 --token abcdef.0123456789abcdef ...
+```
+
+### kinder cp
+
+`kinder cp` provide a topology aware wrapper on docker `docker cp` . Following feature are supported:
```

Comment:
some variant of this would also be pretty sweet upstream, but I'm generally +1 on develop ASAP here for kubeadm testing and filter back something like it based on what we learn from usage, if you're up for that :sweat_smile:

Reply:
I'm definitely up for that!